tests/functional/test_data_completeness.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from botocore.session import Session
+from botocore.loaders import Loader
+from botocore.exceptions import DataNotFoundError
+
+
+class TestDataTypes(object):
```

Comment:
@rayluo That's not what I'm saying.  I don't think you can make a sweeping generalization like "a class-based approach is better."  If you're taking advantage of things that a class gives you (inheritance, instance state, etc) then yes it's better.  If you're not using those things (like the situation here) then it's not any better than a function.

If we ever run into a scenario where a class based approach is *actually* better, and provides immediate benefit, I don't think you'll get any push back from us.

At any rate, here's what I suggest:

* Take @kyleknap and my suggestion.  Be simple and consistent.  Use functions.  This PR is fixing a bug whose fix (which we all agree on) should not be blocked on a discussion on changing how we write test generators.

* Continue this discussion independently of this bug fix.  If you're willing to put in the effort to rework how we write test generators at a more holistic level, then that's certainly something we can discuss.

Sound good?